api/cluster/controller.go

```diff
@@ -223,10 +223,19 @@ func (c *controller) Deploy(ctx context.Context, modelService *models.Service) (
 		return nil, errors.Wrapf(err, fmt.Sprintf("%v (%s)", ErrUnableToCreateInferenceService, isvcName))
 	}
 
-	s, err := c.kserveClient.InferenceServices(modelService.Namespace).Create(spec)
+	// check the cluster to see if the inference service has already been deployed
+	s, err := c.kserveClient.InferenceServices(modelService.Namespace).Get(modelService.Name, metav1.GetOptions{})
```

Comment:
to confirm `modelService.Name` is unique per deployment/version/revision right?

Reply:
Yes, it is. Ref: https://github.com/caraml-dev/merlin/blob/904a8940dc8911c0948bf6452c6670ec9aef0942/api/models/service.go#L60C20-L60C46

Reply:
Thanks Arief for sharing the link! 🙏